Code/Tools/ProjectManager/Source/ProjectsScreen.cpp

```diff
@@ -293,20 +279,26 @@ namespace O3DE::ProjectManager
             for (const ProjectInfo& project : projectsVector)
             {
                 ProjectButton* currentButton = nullptr;
-                if (!m_projectButtons.contains(QDir::toNativeSeparators(project.m_path)))
+                auto projectButtonIter = m_projectButtons.find(QDir::toNativeSeparators(project.m_path));
```

Comment:
If this is stored as QUrl, then the path separators don't need to be taken into account when looking it up
Classes like QUrland AZ::IO::Path takes care of issues related to hashing paths that have different path separators and cases for Windows path and are much better to use in that case than a QString or AZStd::string.

The `QUrl` class has [fromLocalFile](https://doc.qt.io/qt-5/qurl.html#fromLocalFile) and [toLocalFile](https://doc.qt.io/qt-5/qurl.html#toLocalFile) functions for working with system paths.
As the QHash global functions has an overload that uses accepts a QUrl, https://doc.qt.io/qt-5/qhash.html#qHashx, that can be used as the key instead of a QString

Reply:
`project.m_path` is a QString, but I agree we should change it to a path type like QDir or AZ::IO::Path,  QUrl might be more than we need.

Reply:
I started to convert this and other path types to AZ::IO::Path but it ballooned really quickly into a lot of changes so I'm going to address that in a separate PR dedicated to that change alone.